devimint/src/tests.rs

```diff
@@ -2103,6 +2108,13 @@ pub async fn recoverytool_test(dev_fed: DevFed) -> Result<()> {
     let diff = balances_after.mine.immature + balances_after.mine.trusted
         - balances_before.mine.immature
         - balances_before.mine.trusted;
+
+    // We need to wait for a session to be generated to make sure we have the signed
+    // session outcome in our DB. If there ever is another problem here: wait for
+    // fedimintd-0 specifically to acknowledge the session switch. In practice this
+    // should be sufficiently synchronous though.
+    wait_session_outcome(&client, last_tx_session).await?;
```

Comment:
I think this test was previously using a different (incorrect?) approach to awaiting the session outcome by looping until 

```
        if outputs.len() >= fed_utxos_sats.len() {
            break outputs.clone();
        }
```

I haven'f fully considered how, but this might be a bug since the epochs method includes descriptors from spent outputs in addition to UTXOs so we might break too early in this loop.

Now that we're explicitly waiting for the session outcome I think we can get rid of the loop and just call the recoverytool with epochs once.

Reply:
I removed the loop, good catch!